components/coda/coda.app.mjs

```diff
@@ -1,11 +1,307 @@
+import { axios } from "@pipedream/platform";
+
 export default {
   type: "app",
   app: "coda",
-  propDefinitions: {},
+  propDefinitions: {
+    title: {
+      type: "string",
+      label: "Title",
+      description: "Title of the doc. Defaults to `\"Untitled\"`",
+      optional: true,
+    },
+    docId: {
+      type: "string",
+      label: "Doc ID",
+      description: "ID of the Doc",
+      async options({ prevContext }) {
+        const response = await this.listDocs(this, {
+          pageToken: prevContext.nextPageToken,
+        });
+        return this._makeOptionsResponse(response);
+      },
+    },
+    folderId: {
+      type: "string",
+      label: "Folder ID",
+      description: "ID of the folder",
+      optional: true,
+    },
+    tableId: {
+      type: "string",
+      label: "Table ID",
+      description: "ID of the table",
+      async options({
+        docId,
+        prevContext,
+      }) {
+        const response = await this.listTables(this, docId, {
+          pageToken: prevContext.nextPageToken,
+        });
+        return this._makeOptionsResponse(response);
+      },
+    },
+    rowId: {
+      type: "string",
+      label: "Row ID",
+      description: "ID of the row",
+      async options({
+        docId, tableId, prevContext,
+      }) {
+        let counter = 0;
+        const response = await this.findRow(this, docId, tableId, {
+          pageToken: prevContext.nextPageToken,
+          sortBy: "natural",
+        });
+
+        const {
+          options,
+          context,
+        } = this._makeOptionsResponse(response);
+
+        return {
+          options: options
+            .map((row) => ({
+              label: `${counter++}: ${row.label}`,
+              value: row.value,
+            })),
+          context,
```

Comment:
Since the `counter` variable is set to `0` each time this `options` functions is called, each page of rows will have row numbers starting at `0`.

To maintain the counter across `options` invocations, you can pass `counter` through [`context`](https://pipedream.com/docs/components/api/#component-api). For example:

```suggestion
        let { counter = 0 } = prevContext;
        const response = await this.findRow(this, docId, tableId, {
          pageToken: prevContext.nextPageToken,
          sortBy: "natural",
        });

        const {
          options,
          context,
        } = this._makeOptionsResponse(response);

        return {
          options: options
            .map((row) => ({
              label: `${counter++}: ${row.label}`,
              value: row.value,
            })),
          context: {
            ...context,
            counter,
          },
```

Reply:
Addressed